Update partition-data.md

rsarosh opened the pull request:
Trying to keep it short. In this document we are explaining the concept so lets leave  the code out of this document. Creating some more emphasis on certain things which i see users often forget or overlook. Remove some implementation detail about hashing etc. No need to bring it up.

Made the file shorter. Removed the code from this file, the code should go to a new file and a link be provided for the code.

azurecla commented:
Hi __@MCKLMT__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

rsarosh committed:
Update partition-data.md

Trying to keep it short. In this document we are explaining the concept and leaving it the code out of this document. Creating some more emphasis on certain things which i see users often forget or overlook.
Made the file shorter. Removed the code from this file, the code should go to a new file and a link be provided for the code.

mimig1 commented on articles/cosmos-db/partition-data.md:
```diff
@@ -52,10 +52,10 @@ The semantics for partition keys are slightly different to match the semantics o
 | Graph | Custom partition key property | Fixed `id` | 
 | Table | Fixed `PartitionKey` | Fixed `RowKey` | 
 
-Azure Cosmos DB uses hash-based partitioning. When you write an item, Azure Cosmos DB hashes the partition key value and uses the hashed result to determine which partition to store the item in. Azure Cosmos DB stores all items with the same partition key in the same physical partition. The choice of the partition key is an important decision that you have to make at design time. You must pick a  ...
```
Why get rid of this?

mimig1 commented on articles/cosmos-db/partition-data.md:
```diff
@@ -52,10 +52,10 @@ The semantics for partition keys are slightly different to match the semantics o
 
 > [!NOTE]
-> It's a best practice to have a partition key with many distinct values (hundreds to thousands at a minimum).
+> It's a best practice to have a partition key with many distinct values (hundreds to thousands at a minimum). Remember! partition key is a logical concept and partitions are physical partitions. These are two different thing. For a 10 partition system, you can have tens of thousands of partition keys.
```
Please change to "Remember, the partition key is a logical ...", we can't use ! mid sentence. Or we can turn it into a NOTE, or TIP, or IMPORTANT item, which will put it in a colored box on the screen. Let me know your preference.

mimig1 commented on articles/cosmos-db/partition-data.md:
```diff
@@ -68,106 +68,6 @@ Azure Cosmos DB is designed for predictable performance. When you create a conta
 > 
 > 
 
-<a name="designing-for-partitioning"></a>
```
Makes sense about removing the code, but I'm wondering where to put it. Even though you do it slightly differently per data-model, I think keeping all the partition samples in the same article makes sense - otherwise if we start creating a separate doc per task, per data-model we'll have a million articles. So I'd suggest a new topic under How To Guides/Develop and call it Partition in the TOC.

arramac replied:
@rafats I think we really need all the code in this article. As it stands, a lot of customers don't understand how to do this, and having this in a separate article will make it harder to find ...

mimig1 reviewed (changes requested):
See comments on individual changes.
